dotnet/src/Connectors/Connectors.HuggingFace/Core/HuggingFaceMessageApiClient.cs

```diff
@@ -0,0 +1,231 @@
+                content: choice.Message?.Content,
+                modelId: response.Model,
+                innerContent: response,
+                encoding: Encoding.UTF8,
+                metadata: metadata));
+        }
+
+        return chatMessageContents;
+    }
+
+    private static StreamingChatMessageContent GetStreamingChatMessageContentFromStreamResponse(ChatCompletionStreamResponse response, string modelId)
+    {
+        var choice = response.Choices.FirstOrDefault();
+        if (choice is not null)
+        {
+            var metadata = new HuggingFaceChatCompletionMetadata
+            {
+                Id = response.Id,
+                Model = response.Model,
+                @Object = response.Object,
+                SystemFingerPrint = response.SystemFingerprint,
+                Created = response.Created,
+                FinishReason = choice.FinishReason,
+                LogProbs = choice.LogProbs,
+            };
+
+            var streamChat = new StreamingChatMessageContent(
+                choice.Delta?.Role is not null ? new AuthorRole(choice.Delta.Role) : null,
+                choice.Delta?.Content,
+                response,
+                choice.Index,
+                modelId,
+                Encoding.UTF8,
+                metadata);
+
+            return streamChat;
+        }
+
+        throw new KernelException("Unexpected response from model")
+        {
+            Data = { { "ResponseData", response } },
+        };
+    }
+
+    private async IAsyncEnumerable<StreamingChatMessageContent> ProcessChatResponseStreamAsync(Stream stream, string modelId, [EnumeratorCancellation] CancellationToken cancellationToken)
+    {
+        await foreach (var content in this.ParseChatResponseStreamAsync(stream, cancellationToken).ConfigureAwait(false))
+        {
+            yield return GetStreamingChatMessageContentFromStreamResponse(content, modelId);
+        }
+    }
+
+    private ChatCompletionRequest CreateChatRequest(
+        ChatHistory chatHistory,
+        PromptExecutionSettings? promptExecutionSettings)
+    {
+        var huggingFaceExecutionSettings = HuggingFacePromptExecutionSettings.FromExecutionSettings(promptExecutionSettings);
+        huggingFaceExecutionSettings.ModelId ??= this._clientCore.ModelId;
+
+        HuggingFaceClient.ValidateMaxTokens(huggingFaceExecutionSettings.MaxTokens);
+        var request = ChatCompletionRequest.FromChatHistoryAndExecutionSettings(chatHistory, huggingFaceExecutionSettings);
+        return request;
+    }
+
+    private IAsyncEnumerable<ChatCompletionStreamResponse> ParseChatResponseStreamAsync(Stream responseStream, CancellationToken cancellationToken)
+        => SseJsonParser.ParseAsync<ChatCompletionStreamResponse>(responseStream, cancellationToken);
+
+    private Uri GetChatGenerationEndpoint()
+        => new($"{this._clientCore.Endpoint}{this._clientCore.Separator}v1/chat/completions");
```

Comment:
nit: `=> new($"{this._clientCore.Endpoint}/v1/chat/completions");`

Reply:
I need to keep the separator as the 

Uri.ToString() adds `/` when the path is root but doesn't when the path has more details:

![image](https://github.com/microsoft/semantic-kernel/assets/19890735/093238de-aee1-4fc3-b0d1-892b24308320)